src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -100,58 +196,293 @@ public String refreshOwned() {
+        List<Container> repositories = containerDAO.findByUserId(userId);
+        return repositories;
+    }
+
+    @GET
+    @Timed
+    @Path("getAllRegisteredContainers")
+    public List<Container> getAllRegisteredContainers() {
+        List<Container> repositories = containerDAO.findAll();
+        return repositories;
+    }
+
+    @PUT
+    @Timed
+    @UnitOfWork
+    @Path("/shareWithUser")
+    @ApiOperation(value = "User shares a container with a chosen user", notes = "Needs to be fleshed out.")
+    public void shareWithUser(@QueryParam("container_id") Long containerId, @QueryParam("user_id") Long userId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @PUT
+    @Timed
+    @UnitOfWork
+    @Path("/shareWithGroup")
+    @ApiOperation(value = "User shares a container with a chosen group", notes = "Needs to be fleshed out.")
+    public void shareWithGroup(@QueryParam("container_id") Long containerId, @QueryParam("group_id") Long groupId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/getRepo/{userId}/{repository}")
+    @ApiOperation(value = "Fetch repo from quay.io", response = String.class)
+    public String getRepo(@ApiParam(value = "The full path of the repository. e.g. namespace/name") @PathParam("repository") String repo,
+            @ApiParam(value = "user id") @PathParam("userId") long userId) {
+        List<Token> tokens = tokenDAO.findByUserId(userId);
+        StringBuilder builder = new StringBuilder();
+
+        for (Token token : tokens) {
+            if (token.getTokenSource().equals(TokenType.QUAY_IO.toString())) {
+                Optional<String> asString = ResourceUtilities.asString(TARGET_URL + "repository/" + repo, token.getContent(), client);
+
+                if (asString.isPresent()) {
+                    builder.append(asString.get());
+                }
+                builder.append("\n");
+            }
+        }
+
+        return builder.toString();
+    }
+
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/getBuilds")
+    @ApiOperation(value = "Get the list of repository builds.", notes = "For TESTING purposes. Also useful for getting more information about the repository.", response = String.class)
+    public String getBuilds(@QueryParam("repository") String repo, @QueryParam("userId") long userId) {
+
+        List<Token> tokens = tokenDAO.findByUserId(userId);
+        StringBuilder builder = new StringBuilder();
+
+        for (Token token : tokens) {
+            if (token.getTokenSource().equals(TokenType.QUAY_IO.toString())) {
+                Optional<String> asString = ResourceUtilities.asString(TARGET_URL + "repository/" + repo + "/build/", token.getContent(),
+                        client);
+
+                if (asString.isPresent()) {
+                    String json = asString.get();
+
+                    Gson gson = new Gson();
```

Comment:
Add as a to-do, we should probably use a Java API for this rather than raw API calls, although this is sufficient for now.